docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,213 @@
+# Contributing to Devworkspace-Operator
+
+Hello there! thank you for choosing to the contributing to devfile/devworkspace-operator. Navigate through the following to understand more about contributing here.
+
+- [Contributing to Devworkspace-Operator](#contributing-to-devworkspace-operator)
+- [Before You Get Started](#before-you-get-started)
+  - [Code of Conduct](#code-of-conduct)
+  - [For Newcomers](#for-newcomers)
+- [How to Contribute](#how-to-contribute)
+  - [Set up your Local Development Environment](#set-up-your-local-development-environment)
+  - [Testing Changes](#testing-changes)
+  - [Signing-off on Commits](#signing-off-on-commits)
+    (#testing-your-changes)
+  - [Signing-off on Commits](#signing-off-on-commits)
+
+# Before You Get Started
+
+## Code of Conduct
+
+## For Newcomers
+
+# How to Contribute
+
+<!--
+## Prerequisites
+
+Make sure you have the following prerequisites installed on your operating system before you start contributing:
+
+- [Nodejs and npm](https://nodejs.org/en/)
+
+  To verify run:
+
+  ```
+  node -v
+  ```
+
+  ```
+  npm -v
+  ```
+
+- [Gatsby.js](https://www.gatsbyjs.com/)
+
+  To verify run:
+
+  ```
+  gatsby --version
+  ```
+
+**Note:** If you're on a _Windows environment_ then it is highly recommended that you install [Windows Subsystem for Linux (WSL)](https://docs.microsoft.com/en-us/windows/wsl/install) both for performance and ease of use. Refer to the [documentation](https://docs.microsoft.com/en-us/windows/dev-environment/javascript/gatsby-on-wsl) for the installation of _Gatsby.js on WSL_. -->
+
+## Set up your Local Development Environment
+
+Follow the following instructions to start contributing.
+
+**1.** Fork [this](https://github.com/devfile/devworkspace-operator) repository.
+
+**2.** Clone your forked copy of the project.
+
+```
+git clone https://github.com/<your-github-username>/devworkspace-operator.git
+```
+
+**3.** Navigate to the project directory.
+
+```
+cd devworkspace-operator
+```
+
+**4.** Add a reference(remote) to the original repository.
+
+```
+git remote add upstream https://github.com/devfile/devworkspace-operator.git
+```
+
+**5.** Check the remotes for this repository.
+
+```
+git remote -v
+```
+
+**6.** Always take a pull from the upstream repository to your master branch to keep it at par with the main project (updated repository).
```

Comment:
We should avoid focusing on how to set up git repositories for the contributing guide -- assume the user has cloned a suitable repo to their local machine and wants to begin writing code for it. To this end, focus on 
1. What's required
2. What the flow is for starting/debugging/testing

Reply:
Yes, I have added a "testing your changes" section for describing the general workflow of debugging and testing.